libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -45,6 +52,7 @@ internal class RouteAlternativesController constructor(
         if (observers.isEmpty()) {
             nativeRouteAlternativesController.removeObserver(nativeObserver)
         }
+        mainJobControl.job.cancelChildren()
```

Comment:
This should be moved to the if-check above rather than being invoked for each unsubscription.

Reply:
Yes! nice catch